lib/csv/csv.nit

```diff
@@ -15,119 +15,156 @@
 # CSV document handling.
 module csv
 
-# Specifies a CSV format.
-class CsvFormat
+redef class Text
+	# Escape the content of `self` for inclusion in a CSV document
+	private fun escape_to_csv(sep_char, delim_char: Char, eol: String): String do
+		var add_sp = to_escape_csv(sep_char, delim_char, eol)
+		if add_sp == 0 then return to_s
+		var bf = new Buffer.with_cap(add_sp + bytelen)
+		bf.add '"'
+		for i in [0 .. length[ do
+			var c = self[i]
+			if c == delim_char then
+				bf.add c
+			end
+			bf.add c
+		end
+		bf.add '"'
+		return bf.to_s
+	end
+
+	# How many more bytes should be allocated for CSV escaping ?
+	private fun to_escape_csv(sep_char, delim_char: Char, eol: String): Int do
+		var more_ln = 0
+		var ln = length
+		var need_esc = false
+		var fst_eol = eol.first
+		var i = 0
+		while i < ln do
+			var c = self[i]
+			if c == delim_char then more_ln += 1
+			if c == fst_eol then
+				need_esc = true
+				for j in [1 .. eol.length[ do
+					i += 1
+					c = self[i]
+					if c != eol[j] then
+						i -= j
+						need_esc = false
+						break
+					end
+				end
+			end
+			if c == sep_char then need_esc = true
+			i += 1
+		end
+		var more = more_ln * delim_char.u8char_len
+		if need_esc then more += 2
+		return more
+	end
+
+	# Unescape the content of `self` from CSV format to Nit String
+	private fun unescape_csv(delim_char: Char): String do
+		var to_un = to_unescape_csv(delim_char)
+		if to_un == 0 then return to_s
+		var buf = new Buffer.with_cap(bytelen - to_un)
+		var pos = 0
+		var ln = length
+		while pos < ln do
+			var c = self[pos]
+			if c == delim_char then pos += 1
+			buf.add c
+			pos += 1
+		end
+		return buf.to_s
+	end
+
+	# How many bytes should be removed for CSV unescaping ?
+	private fun to_unescape_csv(delim_char: Char): Int do
+		var pos = 0
+		var to_un = 0
+		var ln = length
+		while pos < ln do
+			var c = self[pos]
+			if c == delim_char then
+				pos += 1
+				to_un += 1
+			end
+			pos += 1
+		end
+		return to_un
+	end
+end
+
+# Any class using CSV to write/read data
+class CsvStream
```

Comment:
The only purpose of this class is to share the definitions of `eol`, `delimiter` and `separator`, I can improve the documentation and make it private, this is more of an implementation detail than anything else